http-clients/src/main/java/com/palantir/remoting/http/errors/SerializableErrorErrorDecoder.java

```diff
@@ -115,6 +118,25 @@ private static Exception constructException(
         // Note: If another constructor is added, then we should refactor the construction logic in order to avoid
         // nested try/catch
         try {
+            if (exceptionClass.equals(ClientErrorException.class)
+                    || exceptionClass.equals(ServerErrorException.class)
+                    || exceptionClass.equals(WebApplicationException.class)) {
```

Comment:
Why not do `WebApplicationException.class.isAssignableFrom(exceptionClass)` to cover all of these cases or do you want to explicitly exclude `javax.ws.rs.RedirectionException`?